### PR TITLE
804 - Added required fields. Added display heading toggle. Added validation for only unique tab backgroud colours.

### PR DESCRIPTION
**Pull request by robbdavis**
804 - Added required fields. Added display heading toggle. Added validation for only unique tab backgroud colours.

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_tabs/ucgy_gbl_tabs.module [hunk 1]
```diff
@@ -103,4 +106,30 @@ function ucgy_gbl_tabs_after_build($form, &$form_state) {
...
+
+   // Add each background colour value to the array
+   $colours[] = $field_item['field_uctab_colour'][LANGUAGE_NONE][0]['rgb'];
+
+ }
+
+  // If the number of unique colours is less than the number of colours
+  // then we have a duplicaate and should throw a validation error.
+  if(count(array_unique($colours)) < count($colours)){
+    form_set_error('mail', t('The background colour for each tab must be unique.'));
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_tabs/ucgy_gbl_tabs.module]: Good catch. Copy and past error. Fixed

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_tabs/templates/fieldable-panels-pane--ucgy_gbl_tabs.tpl.php [hunk 2]
```diff
@@ -8,10 +8,13 @@
             <?php if(!empty($content['title']['#value'])||isset($content['field_uctab_description'])): ?>
             <div class="row header">
                 <div class="col-sm-12">
+                  <?php if($content['field_uctab_disable_heading']['#items'][0]['value'] == 0): ?>
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_tabs/templates/fieldable-panels-pane--ucgy_gbl_tabs.tpl.php]: Done

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_tabs/templates/fieldable-panels-pane--ucgy_gbl_tabs.tpl.php [hunk 3]
```diff
@@ -8,10 +8,13 @@
...
                     <h3 style="font-weight: bold;"><?php print render($content['title']['#value']); ?></h3>
-                    <?php if(isset($content['field_uctab_description'])): ?>
+                  <?php endif; ?>
+
+                  <?php if(isset($content['field_uctab_description'])): ?>
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_tabs/templates/fieldable-panels-pane--ucgy_gbl_tabs.tpl.php]: Done

**Comment by taeram** [comment]
Closing this PR on GitHub, but keeping the branch around for historical purposes.

**Pull request closed without merging**